packages/shared/scope-hoisting/src/hoist.js

```diff
@@ -676,7 +788,10 @@ const VISITOR: Visitor<MutableAsset> = {
       .getDependencies()
       .find(dep => dep.moduleSpecifier === path.node.source.value);
 
-    if (dep) dep.symbols.ensure();
+    if (dep) {
+      dep.meta.isES6Module = true;
```

Comment:
> isES6Module meaning "created by an ES6 import"?

Yes, perhaps a different name could be better. I guess I was matching the existing asset meta property.

> This could very well happen with an ESM import by the user and the Node global Buffer replacement (adding a CJS import)

I don't quite follow... That's not an ESM import right?

Reply:
For this asset:
```
const x = require("buffer");
import {y} from "buffer";
```
both `dep.meta.isES6Module` and `dep.meta.isCommonJS ` would be true? Is that a problem?

Reply:
Hmm. I guess this is more of a problem with merging multiple imports into a single dependency object...

This would have already been the case even with just `dep.meta.isCommonJS` though, it just wasn't tracked.